proxy/coco/src/project/checkout.rs

```diff
@@ -78,20 +92,44 @@ where
 
         let mut builder = git2::build::RepoBuilder::new();
         builder.branch(self.project.default_branch());
-        builder.remote_create(|repo, _, url| {
-            let remote = Remote::rad_remote(url, None).create(repo)?;
-            Ok(remote)
-        });
-        let repo = git2::build::RepoBuilder::clone(
-            &mut builder,
-            &LocalUrl::from_urn(self.project.urn(), peer_id).to_string(),
-            &project_path,
-        )?;
 
-        super::set_rad_upstream(&repo, self.project.default_branch())?;
+        let repo = match ownership {
+            Ownership::Local(local) => {
+                builder
+                    .remote_create(|repo, _remote_name, url| repo.remote(config::RAD_REMOTE, url));
+                let url = LocalUrl::from_urn(self.project.urn(), local).to_string();
+                git2::build::RepoBuilder::clone(&mut builder, &url, &project_path)
+            },
+            Ownership::Remote {
+                handle,
+                remote,
+                local,
+            } => {
+                let name = format!("{}@{}", handle, remote);
+                builder.remote_create(move |repo, _remote_name, url| {
+                    Remote::new(url, name.clone()).create(repo)
+                });
+                let remote_url = LocalUrl::from_urn(self.project.urn(), remote).to_string();
+                let repo =
+                    git2::build::RepoBuilder::clone(&mut builder, &remote_url, &project_path)?;
 
-        include::set_include_path(&repo, self.include_path)?;
+                // Create a rad remote and push the default branch so we can set it as the
+                // upstream.
```

Comment:
Why are we pushing to the remote for another peer by default?

Reply:
We're not :) We're pushing for our peer so that they have a copy of the default branch under their `rad` remote. Similar to having the default branch under `origin` when you clone from `GitHub`. This then becomes your version of the default branch.

Reply:
I see, I guess. It's hard to follow which end is which in this code and what the to expected state at the end is for the working copy.

Reply:
Cool, that's good to know. I'll make it clearer by maybe breaking it out or more documentation. Tbh, I could only reason through it by looking at the outcomes locally so I'm not surprised it's not clear :)